main.rb

```diff
@@ -0,0 +1,229 @@
+# rubocop:disable Style/For
+# rubocop:disable Metrics/ModuleLength
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return enum_for unless block_given?
+
+    if instance_of?(Hash)
+      for i in 0..length - 1 do
+        yield keys[i], values[i]
+      end
+    else
+      for i in 0..to_a.length - 1 do
+        yield to_a[i]
+      end
+    end
+    self
+  end
+
+  def my_each_with_index
+    return enum_for unless block_given?
+
+    if instance_of?(Hash)
+      for i in 0..length - 1 do
+        yield keys[i], values[i], i
+      end
+    else
+      for i in 0..to_a.length - 1 do
+        yield to_a[i], i
+      end
+    end
+    self
+  end
```

Comment:
# my_each_with_index
- [ ] `my_each_with_index` when a block is given when self is a hash behaves like #each_with_index calls the given block once for each element in self

     Failure/Error: expect(my_each_with_index_output).to eq(each_with_index_output)

       expected: "Num: [:a, 1], idx: 0\nNum: [:b, 2], idx: 1\nNum: [:c, 3], idx: 2\nNum: [:d, 4], idx: 3\nNum: [:e, 5], idx: 4\n"
            got: "Num: a, idx: 1\nNum: b, idx: 2\nNum: c, idx: 3\nNum: d, idx: 4\nNum: e, idx: 5\n"